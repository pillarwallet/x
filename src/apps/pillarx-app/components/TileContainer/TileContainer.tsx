import { ReactNode } from 'react';

type TileContainerProps = {
  children: ReactNode;
  className?: string;
  id?: string;
};

const TileContainer = ({ children, className, id }: TileContainerProps) => {
  return (
    <div id={id} className={`flex bg-container_grey rounded-2xl ${className}`}>
      {children}
    </div>
  );
};

export default TileContainer;
