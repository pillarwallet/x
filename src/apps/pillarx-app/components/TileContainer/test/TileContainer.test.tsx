import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

// components
import TileContainer from '../TileContainer';

describe('<TileContainer />', () => {
  const singleChild = <div>Div as children</div>;
  const multipleChildren = (
    <>
      <div>First child</div>
      <div>Second child</div>
    </>
  );

  it('renders correctly and matches snapshot', () => {
    const tree = renderer
      .create(
        <>
          <TileContainer>{singleChild}</TileContainer>
          <TileContainer className="custom-class">{singleChild}</TileContainer>
          <TileContainer>{multipleChildren}</TileContainer>
        </>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders the single child correctly', () => {
    const tree = renderer
      .create(<TileContainer>{singleChild}</TileContainer>)
      .toJSON() as ReactTestRendererJSON;
    const singleDivChild = tree.children?.find(
      (child) =>
        typeof child === 'object' &&
        child.type === 'div' &&
        child.children?.includes('Div as children')
    ) as ReactTestRendererJSON;

    expect(singleDivChild).not.toBeNull();
    expect(singleDivChild.type).toBe('div');
    expect(singleDivChild.children).toContain('Div as children');
  });

  it('renders multiple children correctly', () => {
    const tree = renderer
      .create(<TileContainer>{multipleChildren}</TileContainer>)
      .toJSON() as ReactTestRendererJSON;
    const multipleDivsChild = tree.children?.filter(
      (child) => typeof child === 'object' && child.type === 'div'
    ) as ReactTestRendererJSON[];

    expect(multipleDivsChild).toHaveLength(2);

    const firstChild = multipleDivsChild[0];
    const secondChild = multipleDivsChild[1];

    expect(firstChild.children).toContain('First child');
    expect(secondChild.children).toContain('Second child');
  });

  it('applies the default classes', () => {
    const tree = renderer
      .create(<TileContainer>{singleChild}</TileContainer>)
      .toJSON() as ReactTestRendererJSON;
    expect(tree.type).toBe('div');
    expect(tree.props.className).toContain('flex');
    expect(tree.props.className).toContain('bg-container_grey');
    expect(tree.props.className).toContain('rounded-2xl');
  });

  it('applies the custom class', () => {
    const customClass = 'custom-class';
    const tree = renderer
      .create(
        <TileContainer className={customClass}>{singleChild}</TileContainer>
      )
      .toJSON() as ReactTestRendererJSON;
    expect(tree.props.className).toContain('flex');
    expect(tree.props.className).toContain('bg-container_grey');
    expect(tree.props.className).toContain('rounded-2xl');
    expect(tree.props.className).toContain(customClass);
  });
});
