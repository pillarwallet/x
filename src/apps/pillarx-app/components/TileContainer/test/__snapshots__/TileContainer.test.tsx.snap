// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TileContainer /> renders correctly and matches snapshot 1`] = `
[
  <div
    className="flex bg-container_grey rounded-2xl undefined"
  >
    <div>
      Div as children
    </div>
  </div>,
  <div
    className="flex bg-container_grey rounded-2xl custom-class"
  >
    <div>
      Div as children
    </div>
  </div>,
  <div
    className="flex bg-container_grey rounded-2xl undefined"
  >
    <div>
      First child
    </div>
    <div>
      Second child
    </div>
  </div>,
]
`;
